packages/react-storage/src/components/StorageBrowser/Views/Controls/Paginate.tsx

```diff
@@ -1,208 +1,102 @@
-      ariaCurrent = 'page';
-      ariaLabel = `Page ${current}`;
-      children = <PaginateText />;
+    case 'paginate-current': {
+      const page = `${currentPage}`;
+
+      ariaCurrent = 'page' as const;
+      ariaLabel = `Page ${page}`;
+      children = (
+        <SpanElement className={`${BLOCK_NAME}__text`}>{page}</SpanElement>
+      );
       className = `${BLOCK_NAME}__button-current`;
       break;
+    }
     case 'paginate-next':
       ariaLabel = 'Go to next page';
       className = `${BLOCK_NAME}__button-next`;
-      disabled = !hasNext || isLoadingNextPage;
-      onClick = () => handleUpdateState({ type: 'NEXT' });
+      disabled = disableNext;
+      onClick = handleNext;
       children = (
         <IconElement variant={variant} className={`${BLOCK_NAME}__icon`} />
       );
       break;
     case 'paginate-previous':
       ariaLabel = 'Go to previous page';
       className = `${BLOCK_NAME}__button-next`;
-      disabled = !hasPrevious;
-      onClick = () => handleUpdateState({ type: 'PREVIOUS' });
+      disabled = disablePrevious;
+      onClick = handlePrevious;
       children = (
         <IconElement variant={variant} className={`${BLOCK_NAME}__icon`} />
       );
       break;
   }
 
   return {
-    ...props,
-    'aria-current': props['aria-current'] ?? ariaCurrent,
-    'aria-label': props['aria-label'] ?? ariaLabel,
-    children: props.children ?? children,
-    className: props.className ?? className,
-    disabled: props.disabled ?? disabled,
-    onClick: props.onClick ?? onClick,
-    type: props.type ?? 'button',
+    'aria-current': ariaCurrent,
+    'aria-label': ariaLabel,
+    children,
+    className,
+    disabled,
+    onClick,
+    type: 'button',
     variant,
   };
 };
 
-const PaginateButtonControl: typeof ButtonElement = React.forwardRef(
-  function Button({ variant: _variant, ...props }, ref) {
-    const { variant } = React.useContext(PaginateItemContext);
-    const context = useControl({ type: 'PAGINATE' });
-
-    return (
-      <ButtonElement
-        {...getButtonVariantProps(
-          { ref, variant: _variant ?? variant, ...props },
-          context
-        )}
-      />
-    );
-  }
-);
-
-const PaginateItem = (props: PaginateItemProps) => {
-  const { variant } = props;
-
-  return (
-    <PaginateItemContext.Provider key={variant} value={props}>
-      <PaginateItemContainer>
-        <PaginateButtonControl />
-      </PaginateItemContainer>
-    </PaginateItemContext.Provider>
-  );
-};
-
-const CurrentControl = ({ variant }: PaginateItemProps) => (
-  <PaginateItem variant={variant ?? 'paginate-current'} />
-);
-
-const NextControl = ({ variant }: PaginateItemProps) => (
-  <PaginateItem variant={variant ?? 'paginate-next'} />
-);
-
-const PreviousControl = ({ variant }: PaginateItemProps) => (
-  <PaginateItem variant={variant ?? 'paginate-previous'} />
-);
+// all props below are used but eslint is erroring becuase they are passed to a
+// on the
```

Comment:
```suggestion
// all props below are used but eslint is erroring as they are passed to 
// `getButtonVariantProps` instead of a React component
```